packages/config-plugins/src/android/Version.ts

```diff
@@ -34,7 +34,7 @@ export function setVersionName(config: Pick<ExpoConfig, 'version'>, buildGradle:
 }
 
 export function getVersionCode(config: Pick<ExpoConfig, 'android'>) {
-  return config.android?.versionCode ?? 1;
+  return config.android?.versionCode;
```

Comment:
Repeating [comment](https://github.com/expo/expo-cli/pull/3803#discussion_r700187819) here:

After discussing in slack with @EvanBacon, have come to understand that config-plugins access the expoConfig via getConfig. Thus if we set the default values there (which is part of this PR below), we can delete these "default ensuring" functions once this PR is published..